group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupMetadataManager.java

```diff
@@ -1421,16 +1490,19 @@ private CoordinatorResult<ConsumerGroupHeartbeatResponseData, Record> consumerGr
         int memberEpoch
     ) throws ApiException {
         ConsumerGroup group = consumerGroup(groupId);
+        boolean isDynamicMember = instanceId == null;
+        boolean isStaticMemberTemporaryLeave = memberEpoch == LEAVE_GROUP_STATIC_MEMBER_EPOCH;
```

Comment:
Hum... I thought that we could use a boolean like `maybeDowngrade` and set it to true in the relevant places. I think that it would be clearer than duplicating the conditions.